TLM/TLM/Manager/Impl/RoutingManager.cs

```diff
@@ -2342,5 +2346,45 @@ bool laneChangesAllowed
 
             RecalculateAll();
         }
+
+        /// <summary>
+        /// if the direction of the given lanes do not match then similar lane index must be reversed.
+        /// </summary>
+        private bool ShouldReverseSimilarLaneIndex(
+            bool segmentInvert1, NetInfo.Lane laneInfo1, bool startNode1,
+            bool segmentInvert2, NetInfo.Lane laneInfo2, bool startNode2) {
+#if DEBUG
+            bool logRouting = DebugSwitch.Routing.Get(); 
+#else
+            const bool logRouting = false;
+#endif
+            bool both1 = laneInfo1.m_finalDirection == NetInfo.Direction.Both;
+            bool both2 = laneInfo2.m_finalDirection == NetInfo.Direction.Both;
```

Comment:
Is there a need here to check the `Avoid` directions in case one of the lanes is on a station track segment?

Reply:
station tracks have `AvoidForward`/`AvoidBackward` direction. this means their similar lane index will restart if you look at them from the other side. `AvoidForward` will actually be considered backward in here.